File: CPP/map/Find Missing and Repeated Values.cpp
vector<int> findMissingAndRepeatedValues(vector<vector<int>>& grid) {
        unordered_map<int,int>mp;
        int n=grid.size();
        int missing=-1,repeating=-1;
        for(int i=0;i<n;i++){
            for(int j=0;j<grid[0].size();j++){
                mp[grid[i][j]]++;
            }
        }
        for(int i=1;i<=n*n;i++){
            //if (mp[i]==0) missing = i;
            if(mp.count(i)==0) missing=i;
            if(mp[i]>1)repeating=i;
        }
        return {repeating,missing};
    }

File: Python/dictionary/Find Missing and Repeated Values.py
class Solution:
    def findMissingAndRepeatedValues(self, grid: List[List[int]]) -> List[int]:
        thisdict={}

        n=len(grid)

        for i in range(n):
            for j in range(len(grid[0])):
                thisdict[grid[i][j]]=thisdict.get(grid[i][j],0)+1
        
        missing =-1
        repeating=-1

        for i in range(1,n*n+1):
            if i not in thisdict:
                missing = i
            elif thisdict[i]>1:
                repeating=i
        return [repeating,missing]
        
